open_rag_eval/query_generation/document_sources.py

```
@@ -0,0 +1,344 @@
+            if page_key:
+                params["page_key"] = page_key
+
+            post_headers = {
+                'x-api-key': self.api_key,
+            }
+            response = self.session.get(
+                f"{self.api_url}/v2/corpora/{self.corpus_key}/documents",
+                headers=post_headers,
+                params=params,
+                timeout=30
+            )
+
+            if response.status_code != 200:
+                raise IOError(
+                    f"Error listing documents: status={response.status_code}, "
+                    f"response={response.text}"
+                )
+
+            res = response.json()
+
+            for doc in res.get('documents', []):
+                doc_ids.append(doc['id'])
+
+            response_metadata = res.get('metadata', None)
+            if not response_metadata or not response_metadata.get('page_key'):
+                break
+            page_key = response_metadata['page_key']
+
+        return doc_ids
+
+    def get_doc_text(self, doc_id: str) -> str:
+        """
+        Retrieve text content of a document.
+
+        Args:
+            doc_id: Document ID
+
+        Returns:
+            Document text content
+
+        Raises:
+            IOError: If API request fails
+        """
+        post_headers = {
+            'x-api-key': self.api_key,
+        }
+        response = self.session.get(
+            f"{self.api_url}/v2/corpora/{self.corpus_key}/documents/{doc_id}",
+            headers=post_headers,
+            timeout=30
+        )
+
+        if response.status_code != 200:
+            raise IOError(
+                f"Error retrieving document {doc_id}: status={response.status_code}, "
+                f"response={response.text}"
+            )
+
+        res = response.json()
+        return '\n'.join((p['text'] for p in res.get('parts', [])))
+
+    def load_documents(
+        self,
+        min_doc_size: int = 0,
+        num_docs: Optional[int] = None,
```

given how you're using min/max else where, perhaps name this to `max_num_docs`.

btw, the default will cause major issues for large corpora, so you might want to throw an error if num_docs is None and number of documents in the corpus is more than let's say 5000.